### ISSUE TITLE
numeric rules issues

### ISSUE THREAD

**Issue by mshlz** [skaterdav85/validatorjs#413]
I've found some problems with the numeric rules, I've tried to validate an integer field using the `integer` rule, but when we put an array of single number as input, the rule don't catch it. 

Other numeric related rules have similar problems, like `numeric` and `digits`. The last broke the app, because it tries to call `.trim()` in the passed value (array).

This is a minimalist code to reproduce these errors.
```ts
import Validator from 'validatorjs'

const test = () => {

    const data = { array: [2] }
    const rules = ['integer', 'numeric', 'digits']

    rules.forEach(rule => {

        const validation = new Validator(data, { array: rule })

        if (validation.passes()) {
            console.log(`Rule: ${rule} = passes`)
        }
        else {
            console.log(`Rule: ${rule} = errors: `, validation.errors.all())
        }

    })
}

test()
```

I think these numeric rules should have some validation like `!Array.isArray(value)`. 

JS  has some weird behaviors: 😅
```ts

String([2]) // >> '2'
Number([2]) // >> 2
parseInt([2], 10) // >> 2

```

**Comment by mikeerickson** [comment]
@mshlz The crux of this error was in the `digits` rule (other rules correctly handled this situation)

This issue has been resolved in the upcoming 3.5 release
